packages/holocron/src/holocronModule.jsx

```diff
@@ -97,57 +97,46 @@ export default function holocronModule({
   options = {},
 } = {}) {
   return function wrapWithHolocron(WrappedComponent) {
-    class HolocronModuleWrapper extends React.Component {
-      constructor(props) {
-        super(props);
-        this.state = {
-          loadCount: 0,
-          status: 'loading',
-        };
-        this.mounted = false;
-      }
-
-      componentDidMount() {
-        this.mounted = true;
-        this.initiateLoad(0, this.props);
-      }
-
-      // ignoring to support deprecated componentWillReceiveProps.
-      // This needs to be removed to support React17
-      // eslint-disable-next-line camelcase
-      UNSAFE_componentWillReceiveProps(nextProps) {
-        const { loadCount } = this.state;
-        const { props } = this;
-        if (shouldModuleReload && shouldModuleReload(props, nextProps)) {
-          const newLoadCount = loadCount + 1;
-          this.setState({ status: 'loading', loadCount: newLoadCount });
-          this.initiateLoad(newLoadCount, nextProps);
-        }
-      }
-
-      // Ignoring as internal state cannot be checked when unmounted.
-      /* istanbul ignore next */
-      componentWillUnmount() {
-        this.mounted = false;
-      }
+    const HolocronModuleWrapper = (props) => {
+      const [{ loadCount, status }, setState] = useState({ loadCount: 0, status: 'loading' });
+      const isMounted = useRef(false);
+      const prevPropsRef = useRef({});
 
-      initiateLoad(loadCount, frozenProps) {
-        return executeLoadingFunctions({
+      const initiateLoad = (currentLoadCount, frozenProps) => {
+        executeLoadingFunctions({
```

Comment:
```suggestion
        return executeLoadingFunctions({
```
Just to keep it the same as before

Reply:
I don't see why we need return statement here. But, I can add it.